documentation.md

```diff
@@ -0,0 +1,117 @@
+
+#### Dropout Layer
+The implementation of dropout layers within the model's architecture to reduce the generalizable error were considered. This is due to the fact that a dropout layer with randomly with probably equal to the dropout rate, drop nodes from a network between layers. Given the Lipohypertrophy data, the dropout layers experiments with varying dropout rates have shown no success. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/densemodels-ax-dropout-layers.ipynb). 
+
+#### Recall
+To reduce the generalization error, consider varying the pos_weight argument in the loss function. Increasing pos_weight > 1 means there is a heavier penalization (loss) on positives in an attempt to reduce false negatives (a true positive where the model predicts is negative) and improve recall. This experumentation is also used to combat the class imbalance issue. To see this exploration, click [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/pos-weight-exploration.ipynb).
+
+
+### Hyperparameter Optimization - Bayesian Optimization with Ax
+
+The general idea with the Bayesian method is that not every possible parameterization in the defined parameter space is explored.
+The process tunes parameters in a few iterations by building a surrogate model. A surrogate model is a probablistic model that uses an acquisition function to direct the next sample to make up configurations where an improvement over the current best parameterization is likely. Bayesian Optimization with Ax, a recently developed package for optimization by Facebook, relies on information from previous trials to propose better hyperparameters in the next evaluation. The Gaussian process is as follows.
+
+1. Build "smooth" surrogate model using Gaussian processes (initially Gaussian with mean 0 and variance equal to the noise in data. The surrogate model updates to a Gaussian model with mean equal to the estimated mean and updated variance based on the previous trial. 
+2. use the surrogate model to **predict** the next parameterization (estimated mean), out of the remaining ones and quantify uncertainty (updated estimated variance)
+3. combine predictions and estimates to derive an acquisition function and then optimize it to find the best configuration.
+4. Observe outcomes
+
+Fit new surrogate model and repeat process.
+
+As a way to guide the surrogate model in the prediction of the next parameter configuration to explore, aquisition functions are utilizied. 
+- Three common examples include:
+ 1. Probability of Improvement (PI).
+ 2. Expected Improvement (EI).
+ 3. Lower Confidence Bound (LCB)
+ 
+The most common technique used in Bayesian Optimization is the second choice: Expected Improvement. As such, that was used in the processes experimented [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/densenet-optimized.ipynb).
```

Comment:
Suggestion: "The most common technique used in Bayesian Optimization is the second choice: Expected Improvement. As such, that was used for our [experiments](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/master/notebooks/densenet-optimized.ipynb)".